app/uk/gov/hmrc/helptosave/models/EligibilityCheckResult.scala

```diff
@@ -16,14 +16,33 @@
 
 package uk.gov.hmrc.helptosave.models
 
-import play.api.libs.json.{Format, Json}
+import play.api.libs.json.Json._
+import play.api.libs.json._
 
-/**
-  * The result of performing an eligibility check. If the user is eligible
-  * return their details, otherwise return [[None]]
-  */
-case class EligibilityCheckResult(result: Option[UserInfo])
+case class EligibilityCheckResult(result: Either[HtsError, UserInfo])
 
 object EligibilityCheckResult {
-  implicit val eligibilityResultFormat: Format[EligibilityCheckResult] = Json.format[EligibilityCheckResult]
+
+  val reads: Reads[EligibilityCheckResult] = new Reads[EligibilityCheckResult] {
+    override def reads(json: JsValue): JsResult[EligibilityCheckResult] = {
+
+      val result = (json \ "result").get
+
+      val reads = result.asOpt[HtsError].fold(
+        EligibilityCheckResult(Right(result.as[UserInfo]))
```

Comment:
`as` will throw an exception if the result cannot be read as a `UserInfo` - this should result in a `JsError`